lib/checkout_sdk/checkout_utils.rb

```diff
@@ -5,10 +5,19 @@ class CheckoutUtils
     # @return [CheckoutSdk::HttpMetadata]
     def self.map_to_http_metadata(response)
       metadata = CheckoutSdk::HttpMetadata.new
-      # TODO: convert to hash
-      metadata.headers = response[:headers]
-      metadata.status_code = response[:status]
-      metadata.body = response[:body]
+      return metadata if response.nil?
+
+      if response.is_a?(Hash)
+        metadata.headers = response[:headers]
+        metadata.status_code = response[:status]
+        metadata.body = response[:body]
+      end
+
+      if Faraday::Response.name.start_with? response.class.name
```

Comment:
I would change this to `response.class.name.starts_with? Faraday::Response.name` to improve readability